addons/l10n_tw_edi_ecpay/tests/test_edi.py

```diff
@@ -236,6 +237,45 @@ def test_07_fail_data_validation(self):
         with self.assertRaises(UserError):
             send_and_print.action_send_and_print()
 
+    def test_08_invoice_with_downpayment(self):
+        """Ensure downpayment with -ve quantity is normalized for ECPay JSON."""
+        invoice = self.basic_invoice.copy()
+        invoice.write({
+            "invoice_line_ids": [
+                Command.create({
+                    "name": "Downpayment",
+                    "price_unit": 10.0,
+                    "quantity": -1.0,
+                    "tax_ids": self.tax_sale_a,
+                })
```

Comment:
I'd add some commas to minimize future diffs, in case this test needs to be extended in the future

```suggestion
                }),
```